### ISSUE TITLE
Question about token management

### ISSUE THREAD

**Issue by cderv** [dreamRs/rte.data#3]
Why the choice of storing a token by api name ? 

With data.rte-france.com, you can create an application that can register several api ressource and with the same id client and secret (so same token), you can retrieve several ressource. 

Currently, if I use `set_key` and I need to register once for `get_consumption`, and once for  `get_physical_flows`.  However, I register the two ressource in the same app, so one token can get me both. 

Does your package suppose that one resource is register in one application ? 

I think a mechanism closer to the one in httr with a logic of registering an application to get access to some resources can help avoid duplication in options or environment variables.

**Comment by pvictor** [comment]
Hello,
Yep I misread API documentation and get convinced that you need one token per resource...
I can change that but you're implementation seems perfect :)

Victor

**Comment by cderv** [comment]
Are you talking about https://github.com/cderv/opendatarte ? 

I resurrected it this week end. It has been staled for 2 years but now it is working as expected and a first MVP. 
I like the token management I manage to implement. You only need to do it once per session per application. Or you can even store the token for future sessions. 

However, I did not yet implement all the request function by API to get 
* either a raw list (parsed json)
* either a table with the data
* either a plot for the data (this feature is so cool !) 

Do you plan to keep on going developing this 📦  with new API and enpoints ? 
We may merged our efforts... 

I am currently discussing with the team maintaining the RTE API in order to evaluate how it would be possible to access some documentation to generate automatically all functions in the R package from the API documentation. I think it is possible. last little project of mine ☺️

**Comment by pvictor** [comment]
Oups didn't see it before 🙈 
Yep good work :)

Why I started this package : 

* We used a resource for work (NTC)
* Found the data very interesting at a personal level
* Wanted to play with [{crul}](https://github.com/ropensci/crul)
* Fun with plots

Main result is here : http://shinyapps.dreamrs.fr/rte-data/

I don't have real plan for this package, I won't release on CRAN for example. And not sure to have time to develop methods for new resources. In term of usage I only had one request (not long ago), but never "advertise" for the package ( except the app), a release on CRAN can bring more exposure.

If you plan to publish to CRAN, you can go for it. This package can be a nice showcase for Rte.
And if you're interested in the `autoplot` methods I can make a PR, glad you like this feature, I think it's nice for users to see what's in the data (there is a `ggplot2` method and an interactive one (with {htmlwidgets}), but probably the ggplot2 one is enough).

You can be interested in data in `data-raw` too.

If you can manage to generate R functions from API documentation, it will be gold !

**Comment by cderv** [comment]
Thanks good to know ! 
I’ll see where it leads me. 

And don’t worry I made this for internal use. I told you once long time ago in the subway  😅
I have just made it public this week end after some cleaning. So you would not have been able to see it before. 
httr was lacking credential grant feature for one year. Now it is ok. 

I’ll look into autoplot and other helpers function. If some are useful to reuse, i will tell you. 

I’ll close for now. Thanks !
